infra/packages/shared/src/index.ts

```diff
@@ -42,3 +42,101 @@ export const CLOUDWATCH_KINESIS_STREAM_ROLE_ARN =
+export const DOCUMENT_STORAGE_SERVICE_URL = 'DOCUMENT_STORAGE_SERVICE_URL';
+export const CONNECTION_GATEWAY_URL = 'CONNECTION_GATEWAY_URL';
+export const DOCUMENT_COGNITION_SERVICE_URL = 'DOCUMENT_COGNITION_SERVICE_URL';
+export const LEXICAL_SERVICE_URL = 'LEXICAL_SERVICE_URL';
+export const SEARCH_SERVICE_URL = 'SEARCH_SERVICE_URL';
+
+type MacroUrlName = SYNC_SERVICE_URL
+| COMMS_SERVICE_URL
+| EMAIL_SERVICE_URL
+| PROPERTIES_SERVICE_URL
+| METERING_SERVICE_URL
+| STATIC_FILE_SERVICE_URL
+| ORGANIZATION_SERVICE_URL
+| NOTIFICATION_SERVICE_URL
+| AUTHENTICATION_SERVICE_URL
+| DOCUMENT_STORAGE_SERVICE_URL
+| CONNECTION_GATEWAY_URL
+| DOCUMENT_COGNITION_SERVICE_URL
+| LEXICAL_SERVICE_URL
+| SEARCH_SERVICE_URL;
+
+/*
+ * for service named "foo" it can look like
+ * foo.macro.com
+ * foo-service.macro.com
+ * foo-service-prod.macro.com
+ * some-other-thing.macro.com
+ */
+
+const PROD_URLS = {
+  [SYNC_SERVICE_URL]: 'https://sync-service-prod.macroverse.workers.dev',
+  [COMMS_SERVICE_URL]: 'https://comms-service.macro.com',
+  [EMAIL_SERVICE_URL]: 'https://email-service.macro.com',
+  [PROPERTIES_SERVICE_URL]: 'https://properties-service.macro.com',
+  [METERING_SERVICE_URL]: 'https://metering.macro.com',
+  [STATIC_FILE_SERVICE_URL]: 'https://static-file-service.macro.com',
+  [ORGANIZATION_SERVICE_URL]: 'https://organization-service.macro.com',
+  [NOTIFICATION_SERVICE_URL]: 'https://notifications.macro.com',
+  [AUTHENTICATION_SERVICE_URL]: 'https://auth-service.macro.com',
+  [DOCUMENT_STORAGE_SERVICE_URL]: 'https://cloud-storage.macro.com',
+  [CONNECTION_GATEWAY_URL]: 'https://connection-gateway.macro.com',
+  [DOCUMENT_COGNITION_SERVICE_URL]: 'https://document-cognition.macro.com',
+  [LEXICAL_SERVICE_URL]: 'https://lexical-service-prod.macroverse.workers.dev',
+  [SEARCH_SERVICE_URL]: 'https://search-service.macro.com',
+}
+
+const DEV_URLS = {
+  [SYNC_SERVICE_URL]: 'https://sync-service-dev3.macroverse.workers.dev',
+  [COMMS_SERVICE_URL]: 'https://comms-service-dev.macro.com',
+  [EMAIL_SERVICE_URL]: 'https://email-service-dev.macro.com',
+  [PROPERTIES_SERVICE_URL]: 'https://properties-service-dev.macro.com',
+  [METERING_SERVICE_URL]: 'https://metering-dev.macro.com',
+  [STATIC_FILE_SERVICE_URL]: 'https://static-file-service-dev.macro.com',
+  [ORGANIZATION_SERVICE_URL]: 'https://organization-service-dev.macro.com',
+  [NOTIFICATION_SERVICE_URL]: 'https://notifications-dev.macro.com',
+  [AUTHENTICATION_SERVICE_URL]: 'https://auth-service-dev.macro.com',
+  [DOCUMENT_STORAGE_SERVICE_URL]: 'https://cloud-storage-dev.macro.com',
+  [CONNECTION_GATEWAY_URL]: 'https://connection-gateway-dev.macro.com',
+  [DOCUMENT_COGNITION_SERVICE_URL]: 'https://document-cognition-dev.macro.com',
+  [LEXICAL_SERVICE_URL]: 'https://lexical-service-dev.macroverse.workers.dev',
+  [SEARCH_SERVICE_URL]: 'https://search-service-dev.macro.com',
+}
+
+const URLS = stack === STACK_PROD ? PROD_URLS : DEV_URLS;
```

Comment:
This will break if we ever add a staging environment. I think URLS should be a function that can throw an error if you provide an unexpected stack name.

Reply:
see signal for defining cost